src/shared/utils/__tests__/traverseAllChildren-test.js

```diff
@@ -476,4 +476,45 @@ describe('traverseAllChildren', function() {
     }
   });
 
+  it('should warn about using objects', function() {
+    spyOn(console, 'error');
+
+    var instance = (
+      <div>
+        {{ a: 1, b: 2 }}
+      </div>
+    );
+
+    expect(function() {
+      traverseAllChildren(instance.props.children, function() {});
+    }).toThrow(
+      'Invariant Violation: Objects are not valid as a React child ' +
+      '(found object with keys {a, b}). ' +
+      'If you meant to render a collection of children, use an ' +
+      'array instead or wrap the object using ' +
+      'React.addons.createFragment(object).'
+    )
+  });
+
+  it('should warn about using date objects', function() {
+    spyOn(console, 'error');
+
+    var instance = (
+      <div>
+        {new Date('2015-09-12T17:00:00Z')}
+      </div>
+    );
+
+    try {
+      traverseAllChildren(instance.props.children, function() {});
+      expect("here").toBe("not reached");
+    } catch (ex) {
+      // Can't fuzzy match on error with toThrow
```

Comment:
I tried this in https://github.com/glenjamin/react/commit/7bb820fa55011388ff1dcc8ec49c44f76f22ccde - as far as I can tell this is a Jasmine 2.0 feature that isn't available the vendored jasmine 1.3 jest contains.